### PR TITLE
Add: listen on random local port

### PR DESCRIPTION
**Pull request by AlexStocks**
1 udp/tcp server listen on random local port;
2 using perrors.WithStack in log

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -67,9 +68,9 @@ func newServer(t EndPointType, opts ...ServerOption) *server {
 
 	s.init(opts...)
 
-	if s.addr == "" {
-		panic(fmt.Sprintf("@addr:%s", s.addr))
-	}
+	//if len(s.addr) == 0 {
+	//	panic(fmt.Sprintf("@addr:%s", s.addr))
+	//}
```
Comment (zouyx) [inline server.go:73]: 不用，建议delete

File: server.go [hunk 2]
```diff
@@ -357,7 +372,7 @@ func (s *server) runWSEventLoop(newSession NewSessionCallback) {
 		s.lock.Unlock()
 		err = server.Serve(s.streamListener)
 		if err != nil {
-			log.Errorf("http.server.Serve(addr{%s}) = err{%+v}", s.addr, err)
+			log.Errorf("http.server.Serve(addr{%s}) = err{%+v}", s.addr, perrors.WithStack(err))
 			// panic(err)
```
Comment (zouyx) [inline server.go:376]: 同上

File: server.go (hunk 1 above)
Comment (AlexStocks) [reply server.go]: done

File: server.go (hunk 2 above)
Comment (AlexStocks) [reply server.go]: done

**Comment by wongoo** [comment]
```bash
go fmt ./... && [[ -z `git status -s` ]]
session.go
The command "go fmt ./... && [[ -z `git status -s` ]]" exited with 1.
```

pls format the code session.go

**Comment by AlexStocks** [comment]
> ```shell
> go fmt ./... && [[ -z `git status -s` ]]
> ```

done

**Pull request merged**